skipList/skipList/SkipList.cs

```diff
@@ -0,0 +1,409 @@
+﻿// <copyright file="SkipList.cs" company="Kalinin Andrew">
+// Copyright (c) Kalinin Andrew. All rights reserved.
+// </copyright>
+
+namespace SkipList;
+
+using System.Collections;
+using System.Runtime.ExceptionServices;
+
+/// <summary>
+/// An implementation of a skip list that supports the IListT interface.
+/// </summary>
+/// <typeparam name="T">The type of elements in the collection.</typeparam>
+public class SkipList<T> : IList<T>
+{
+    private const int MaxLevel = 32;
+    private const double Probability = 0.5;
+
+    private readonly Random random = new Random();
+    private readonly IComparer<T> valueComparer;
+    private readonly Node head = new Node(default, MaxLevel);
```

Comment:
```suggestion
    private readonly Node head = new(default, MaxLevel);
```
Но тут всё равно nullability ругается, а надо, чтобы не ругалось